backend/src/app/auth.py

```diff
@@ -0,0 +1,17 @@
+import app
+
+from flask import jsonify, g, request
+from functools import wraps
+
+
+def authenticate(f):
+  @wraps(f)
+  def decorated_function(*args, **kwargs):
+    auth_service = app.services.auth_service()
+    (user, err) = auth_service.authenticate_request(request)
+    if user:
+      return f(*args,**kwargs)
+    else:
```

Comment:
can you add a test for when a user isn't authenticated?

Reply:
Adding tests in a separate PR